strings-csharp/SubstringInString/SubstringInString/SubstringSearchMethods.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System.Text.RegularExpressions;
+
+public static partial class SubstringSearchMethods
+{
+    public static List<int> FindAllIndexesWithSubstring(string input, string search)
+    {
+        var indexes = new List<int>();
+
+        for (var i = 0; i <= input.Length - search.Length; i++)
+        {
+            if (input.Substring(i, search.Length) == search)
+            {
+                indexes.Add(i);
+            }
+        }
+
+        return indexes;
+    }
+    public static void PrintResult(List<int> result)
+    {
+        Console.WriteLine("Indexes where the substring is found: " + string.Join(", ", result));
+    }
+    public static List<int> FindAllIndexesWithSpan(string input, string search)
+    {
+        var indexes = new List<int>();
+        var inputSpan = input.AsSpan();
+        var searchSpan = search.AsSpan();
+
+        for (int i = 0; i <= input.Length - search.Length; i++)
+        {
+            if (inputSpan.Slice(i, search.Length).SequenceEqual(searchSpan))
+            {
+                indexes.Add(i);
+            }
+        }
+
+        return indexes;
+    }
+    public static List<int> FindAllIndexesWithIndexOf(string input, string search)
+    {
+        var indexes = new List<int>();
+        var startIndex = 0;
+
+        do
+        {
+            startIndex = input.IndexOf(search, startIndex);
+            if (startIndex != -1)
+            {
+                indexes.Add(startIndex);
+                startIndex++;
+            }
+        }
+        while (startIndex != -1);
+
+        return indexes;
+    }
+    public static List<int> FindAllIndexesWithRegex(string input, string search)
+    {
+        var indexes = new List<int>();
+        var regex = new Regex(@"\b" + Regex.Escape(search) + @"\b", RegexOptions.IgnoreCase);
+
+        for (int i = 0; i < input.Length - search.Length + 1; i++)
+        {
+            if (regex.IsMatch(input.AsSpan(i, search.Length)))
+            {
+                indexes.Add(i);
+            }
+        }
+
+        return indexes;
+    }
```

Comment:
My suggestion was that this method take a Regex as an input parameter. We can't do what we are doing here and then benchmark it. We are creating a new Regex on every method call. Also, we are doing something that the other methods are not, which is performing a case insensitive match. The other methods are performing case sensitive match.
We need to change the method signature here to be a caller provided Regex. This way in the benchmark we only create the Regex once, either as a source generated regex or using the Compiled option (both are about equal in performance, one has a higher cost on instantiation, but won't matter in the benchmark).